From 5e0b59ee5f1507a522c0aba33e4e6853562281eb Mon Sep 17 00:00:00 2001
From: Sam Saffron <sam.saffron@gmail.com>
Date: Sat, 22 Nov 2025 17:07:21 +1100
Subject: [PATCH] add icons

---
 hyprland-share-picker/mainpicker.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/hyprland-share-picker/mainpicker.cpp b/hyprland-share-picker/mainpicker.cpp
index b6c33bc..e615002 100644
--- a/hyprland-share-picker/mainpicker.cpp
+++ b/hyprland-share-picker/mainpicker.cpp
@@ -50,7 +50,13 @@ void MainPicker::init(const std::vector<SWindowEntry>& windows, bool allowToken)
     // Populate windows
     for (const auto& win : windows) {
         QString text = QString::fromStdString(win.clazz + ": " + win.name);
-        auto item = new QListWidgetItem(text);
+        
+        QString clazz = QString::fromStdString(win.clazz);
+        QIcon icon = QIcon::fromTheme(clazz);
+        if (icon.isNull())
+            icon = QIcon::fromTheme(clazz.toLower());
+            
+        auto item = new QListWidgetItem(icon, text);
         ui->windowList->addItem(item);
         m_windowMap[item] = win.id;
         m_windowHandleMap[item] = win.handle;
